src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,638 @@
+    lastValue = values[i];
+  }
+  segments.push({ left: lastValue, right: max });
+
+  return segments;
+}
+
+const TrackSegment = ({
+  leftPercent,
+  rightPercent,
+}: {
+  leftPercent: number;
+  rightPercent: number;
+}) => {
+  return (
+    <div
+      className='iui-slider-track'
+      style={{ left: `${leftPercent}%`, right: `${rightPercent}%` }}
+    />
+  );
+};
+
+/** Track is component shown a colored segments above Rail. */
+const Track = () => {
+  const { railDomain, trackDisplayMode, values } = useSliderContext();
+  const [currentValues, setCurrentValues] = React.useState(
+    [...values].sort((a, b) => a - b),
+  );
+  React.useEffect(() => {
+    const newValues = [...values];
+    newValues.sort((a, b) => a - b);
+    setCurrentValues(newValues);
+  }, [values]);
+
+  const segments = React.useMemo(
+    () => generateSegments(currentValues, railDomain.min, railDomain.max),
+    [currentValues, railDomain],
+  );
+
+  return (
+    <>
+      {'none' !== trackDisplayMode &&
+        segments.map((segment, index) => {
+          const leftPercent =
+            (100.0 * (segment.left - railDomain.min)) /
+            (railDomain.max - railDomain.min);
+          let rightPercent =
+            (100.0 * (segment.right - railDomain.min)) /
+            (railDomain.max - railDomain.min);
+          rightPercent = 100.0 - rightPercent;
+          if ('odd-segments' === trackDisplayMode && 0 !== (index + 1) % 2) {
+            return (
+              <TrackSegment
+                key={index}
+                leftPercent={leftPercent}
+                rightPercent={rightPercent}
+              />
+            );
+          } else if (
+            'even-segments' === trackDisplayMode &&
+            0 === (index + 1) % 2
+          ) {
+            return (
+              <TrackSegment
+                key={index}
+                leftPercent={leftPercent}
+                rightPercent={rightPercent}
+              />
+            );
+          }
+          return null;
+        })}
+    </>
+  );
+};
+
+/** Thumb is a local component used to show and modify the values maintained by the [[Slider]].
+ * Only one Thumb can be active at a time. A Thumb is made active when the user selects
+ * it with pointer. Whenever a Thumb is active, focused, or hovered its tooltip is shown. */
+const Thumb = ({ index, isActive }: { index: number; isActive: boolean }) => {
+  const {
+    hideTooltip,
+    railDomain,
+    setThumbMoveIndex,
+    toolTipFunction,
+    step,
+    values,
+    setThumbValue,
+    thumbMode,
+    toolTipPosition,
+  } = useSliderContext();
+  const thumbRef = React.useRef<HTMLDivElement>(null);
+  const handleOnKeyDown = React.useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>) => {
+      const [minVal, maxVal] = getAllowableThumbRange(
+        thumbMode,
+        values,
+        index,
+        railDomain,
+        step,
+      );
+
+      if (event.key === 'ArrowLeft') {
```

Comment:
I think using `switch` here would look a little bit better...

Reply:
Converted to switch.